specreduce/extract.py

```diff
@@ -478,10 +618,22 @@ def __call__(self, image=None, trace_object=None,
         disp_axis = disp_axis if disp_axis is not None else self.disp_axis
         crossdisp_axis = crossdisp_axis if crossdisp_axis is not None else self.crossdisp_axis
         bkgrd_prof = bkgrd_prof if bkgrd_prof is not None else self.bkgrd_prof
+        spatial_profile = (spatial_profile if spatial_profile is not None else
+                           self.spatial_profile)
+        n_bins_self_profile = (n_bins_self_profile if n_bins_self_profile is not
+                               None else self.n_bins_self_profile)
+        interp_degree_self_profile = (interp_degree_self_profile if
+                                      interp_degree_self_profile is not None
+                                      else self.interp_degree_self_profile)
         variance = variance if variance is not None else self.variance
         mask = mask if mask is not None else self.mask
         unit = unit if unit is not None else self.unit
 
+        # make sure a valid profile type was specified:
+        spatial_profile = spatial_profile.lower()
+        if spatial_profile not in ['gaussian', 'self_profile']:
+            raise ValueError("spatial_profile must be 'gaussian' or 'self_profile'")
```

Comment:
Would this help future-proof for when we might add more valid options?

```suggestion
        spatial_profile_choices = ('gaussian', 'self_profile')
        if spatial_profile not in spatial_profile_choices:
            raise ValueError(f"spatial_profile must be one of {', '.join(spatial_profile_choices)}")
```